src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/calendar/DueDateCalculator.java

```diff
@@ -2,47 +2,73 @@
 
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.CamundaValue;
 import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.camunda.ConfigurationDmnEvaluationResponse;
 
 import java.time.LocalDateTime;
 import java.util.List;
 import java.util.Optional;
 
+import static uk.gov.hmcts.reform.wataskmanagementapi.services.calendar.DateType.DUE_DATE;
+
 @Slf4j
 @Component
 public class DueDateCalculator implements DateCalculator {
 
     @Override
-    public boolean supports(List<ConfigurationDmnEvaluationResponse> dueDateProperties, boolean isReconfigureRequest) {
-        return Optional.ofNullable(getProperty(dueDateProperties, DUE_DATE)).isPresent()
+    public boolean supports(
+        List<ConfigurationDmnEvaluationResponse> dueDateProperties,
+        DateType dateType,
+        boolean isReconfigureRequest) {
+
+        return DUE_DATE == dateType
+            && Optional.ofNullable(getProperty(dueDateProperties, DUE_DATE.getType())).isPresent()
             && !isReconfigureRequest;
     }
 
     @Override
-    public LocalDateTime calculateDueDate(List<ConfigurationDmnEvaluationResponse> dueDateProperties) {
-        var dueDateResponse = getProperty(dueDateProperties, DUE_DATE);
+    public ConfigurationDmnEvaluationResponse calculateDate(
+        List<ConfigurationDmnEvaluationResponse> dueDateProperties,
+        DateType dateType) {
+        var dueDateResponse = getProperty(dueDateProperties, DUE_DATE.getType());
         var dueDateTimeResponse = getProperty(dueDateProperties, DUE_DATE_TIME);
+        return calculatedDate(dateType, dueDateResponse, dueDateTimeResponse);
+    }
+
+    ConfigurationDmnEvaluationResponse calculatedDate(
```

Comment:
This method should be protected visibility and rest of the methods here are private